core/src/main/scala/org/apache/spark/serializer/SerializerManager.scala

```diff
@@ -103,6 +107,20 @@ private[spark] class SerializerManager(defaultSerializer: Serializer, conf: Spar
   }
 
   /**
+   * Wrap an input stream for encryption if shuffle encryption is enabled
+   */
+  def wrapForEncryption(s: InputStream): InputStream = {
```

Comment:
How about having a "wrap" method that just sets up everything for you? That avoids having the callers in a bunch of places call both `wrapForEncryption` and `wrapForCompression` and have to worry about doing it in the right order.

I'd prefer if the existing `wrapFor...` methods became private or went away, but if they make testing easier, then you can keep them.